### PR TITLE
add "Not Found" error for ipset set & entry delete

### LINKED ISSUE #57318: should ignore the "Not Found error" when ensure ipset set or entry are deleted
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

should ignore the "Not Found error" when ensure ipset set or entry are deleted.

**What happened**:

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig network

/area ipvs

### PR DESCRIPTION
**Pull request by m1093782566**
**What this PR does / why we need it**:

Add "not found error" for ipset set and entry delete. For example,

```
[root@SHA1000130405 k8s-yml]# ipset list
Name: KUBE-LOOP-BACK
Type: hash:ip,port,ip
Revision: 2
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 16776
References: 1
Members:
172.17.0.2,udp:53,172.17.0.2
172.17.0.2,tcp:53,172.17.0.2
100.106.89.164,tcp:6443,100.106.89.164

[root@SHA1000130405 k8s-yml]# ipset delete foo
ipset v6.19: The set with the given name does not exist

[root@SHA1000130405 k8s-yml]# ipset del KUBE-LOOP-BACK 1.2.3.4,tcp:80,1.2.3.4
ipset v6.19: Syntax error: Third element is missing from 1.2.3.4,tcp:80,1.2.3.4.
```
Just need to filter the error message "does not exist" or "element is missing". We should ignore these error when try to make sure they are deleted.

**Which issue(s) this PR fixes**:
Fixes #57318

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/util/ipset/ipset.go [hunk 1]
```diff
@@ -322,4 +322,20 @@ func validatePortRange(portRange string) bool {
 	return true
 }
 
+// IsNotFoundError returns true if the error indicates "not found".  It parses
+// the error string looking for known values, which is imperfect but works in
+// practice.
```
Comment (brendandburns) [inline pkg/util/ipset/ipset.go]: is there a reference to how these strings can be obtained in the future if something changes?

File: pkg/util/ipset/ipset.go [hunk 2]
```diff
@@ -322,4 +322,20 @@ func validatePortRange(portRange string) bool {
...
+func IsNotFoundError(err error) bool {
+	es := err.Error()
+	if strings.Contains(es, "does not exist") {
```
Comment (brendandburns) [inline pkg/util/ipset/ipset.go]: is there a way we can write a test to detect if these strings change?

File: pkg/util/ipset/ipset.go (hunk 1 above)
Comment (m1093782566) [reply pkg/util/ipset/ipset.go]: xref: 

https://github.com/Olipro/ipset/blob/master/lib/errcode.c#L32-L33

https://github.com/Olipro/ipset/blob/master/lib/parse.c#L1904

https://github.com/Olipro/ipset/blob/master/lib/parse.c#L1925

File: pkg/util/ipset/ipset.go (hunk 2 above)
Comment (m1093782566) [reply pkg/util/ipset/ipset.go]: > is there a way we can write a test to detect if these strings change?

I am not sure fully understand your meaning. Do you mean write a test to wath the ipset message string?

These strings are very stable and have not changed for many years.

File: pkg/util/ipset/ipset.go (hunk 1 above)
Comment (m1093782566) [reply pkg/util/ipset/ipset.go]: I added these reference now, PTAL.

**Pull request merged**